scripts/validate_string_concatenation.py

```diff
@@ -1,129 +1,357 @@
 #!/usr/bin/env python
 """
-GH #30454
+Unwanted patterns test cases.
 
-Check where there is a string that needs to be concatenated.
+The reason this file exist despite the fact we already have
+`ci/code_checks.sh`,
+(see https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh)
 
-This is necessary after black formating,
-where for example black transforms this:
-
->>> foo = (
-...         "bar "
-...         "baz"
-...     )
-
-into this:
-
->>> foo = ("bar " "baz")
-
-Black is not considering this as an
-issue (see issue https://github.com/psf/black/issues/1051),
-so we are checking it here.
+is that some of the test cases are more complex/imposible to validate via regex.
+So this file is somewhat an extensions to `ci/code_checks.sh`
 """
 
 import argparse
 import os
 import sys
 import token
 import tokenize
-from typing import Generator, List, Tuple
+from typing import IO, Callable, Generator, List, Tuple
 
 FILE_EXTENSIONS_TO_CHECK = (".py", ".pyx", ".pyx.ini", ".pxd")
 
 
-def main(source_path: str, output_format: str) -> bool:
+def bare_pytest_raises(file_obj: IO) -> Generator[Tuple[int, str], None, None]:
+    """
+    Test Case for bare pytest raises.
+
+    For example, this is wrong:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    And this is what we want instead:
+
+    >>> with pytest.raise(ValueError, match="foo"):
+    ...     # Some code that raises ValueError
+
+    Parameters
+    ----------
+    file_obj : IO
+        File object to operate on.
```

Comment:
Really not sure that this description is good.

Any thoughts?

Reply:
Could be something like "File-like object containing the Python code to validate."